tests/integration/plugins/test_npm.py

```diff
@@ -166,3 +183,69 @@ def test_npm_plugin_include_node(new_dir, partitions):
     # try to use bundled Node.js to execute the script
     output = subprocess.check_output([str(node_path), str(binary)], text=True)
     assert output == "hello world\n"
+
+
+@pytest.mark.slow
+def test_npm_plugin_get_file_list(create_fake_package_with_node, new_dir, partitions):
+    parts = create_fake_package_with_node()
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_npm_plugin",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_file_list()
```

Comment:
I think you should check at least some of the files (the values), in addition to the package names and versions

Reply:
Will do.

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/205557378f9aa07cbaa47c025c1b778dff3b2b0b